---
title: "*Acinetobacter baummanni* OXA Gene Annual Report"
output: pdf_document
---
---
title: "*Acinetobacter baummanni* OXA Gene Annual Report"
output:
  pdf_document: default
  html_document: default
---

```{r load-packages, include=FALSE}
library(tinytex)
library(readr)
library(raster)
library(dplyr)
library(ggplot2)
library(tidyr)
library(knitr)
library(shiny)
library(scales)
library(tidyverse)
library(gridExtra)
library(lubridate)
#library(table1)
#library(kableExtra)
library(sf)
library(ggbreak)
library(gghighlight)
library(flextable)
library(gtsummary)
```

```{r, include=FALSE}
#asking user to select files 
line_list_2019 <- file.choose()
line_list_2020 <- file.choose()
line_list_2021 <- file.choose()
line_list_2022 <- file.choose()
line_list_2023 <- file.choose()
CRO_WGS <- file.choose()
#CRO_WGS_2022 <- file.choose()
NCBI_genes <- file.choose() #remove ","SRS*" in isolate identifiers in excel prior to uploading
#choose SDE_USDC_CENSUS_VA_COUNTY excel file in downloads or on thumb drive                                                                                                                         
VA_map <- file.choose()
```

```{r, include=FALSE}
#open the data sheet from the separate csv files 
linelist_2023 <- read_csv(line_list_2023)
linelist_2022 <- read_csv(line_list_2022)
linelist_2021 <- read_csv(line_list_2021)
linelist_2020 <- read_csv(line_list_2020)
linelist_2019 <- read_csv(line_list_2019)
ncbi <- read_csv(NCBI_genes)
wgs <- read_csv(CRO_WGS)
#wgs2022 <- read_csv(CRO_WGS_2022)
VA_map <- read_sf(VA_map, quiet = TRUE)
```

```{r, include=FALSE}
#define color palettes
Palette <- c("#1F77B4", "#AEC7EB", "#FF7F0E", "#FFBB78", "#2CA02C", 
             "#98DF8A", "#D62728", "#9467BD", "#C5B0D5", "#8C564B", "#C49C94", 
             "#E377C2", "#F7B6D2", "#7F7F7F", "#C7C7C7", "#FFD966", "#DBDB8D", 
             "#17BECF", "#9EDAE5")

OXA_Palette <- c('OXA-23' = "#1F77B4", 
                 'OXA-24' = "#FF7F0E", 
                 'OXA-72' = "#FFBB78",
                 'OXA-23 & OXA-72' = "#AEC7EB",
                 'Other OXA' = "#2CA02C", 
                 "No OXA Genes" = "#7F7F7F")

Cluster_Palette <- c("VA_AB_OXA24_220914 \n or VA_AB_OXA24_220120" = "#1F77B4",
                   "VA_AB_OXA23_240205" = "#AEC7EB",
                   "VA_AB_OXA23OXA24_231117" = "#FF7F0E",
                   "VA_AB_OXA23_230424" = "#FFBB78",
                   "VA_AB_OXA24_230526" = "#98DF8A",
                   "VA_AB_OXA24_231024" = "#D62728",
                   "VA_AB_OXA24_231119" = "#9467BD",
                   "VA_AB_OXA23_220720" = "#C5B0D5", 
                   "VA_AB_OXA23_230626" = "#8C564B",
                   "VA_AB_OXA23_230424" = "#C49C94",
                   "VA_AB_OXA23_230125" = "#E377C2",
                   "VA_AB_OXA23NDM_240106" = "#F7B6D2",
                   "VA_AB_OXA23_221103" = "#7F7F7F",
                   "VA_AB_OXA23_230424-2" = "#FFD966")
```

```{r, warning=FALSE, include=FALSE}
# Cleaning Columns
names(linelist_2023) <- linelist_2023[1,]
linelist_2023 <- linelist_2023[-1,] # Rename column names as first row data
linelist_2023 <- linelist_2023 %>% #Selecting and renaming columns
  dplyr::select('DCLS ID', 'Submitter', 'OXA-23', 'OXA-24/40', 'OXA-58', 
                'Date of collection', 'DCLS - ID confirmation', 
                'Date received at DCLS', 'Source', 'OXA-23-Ref', 'OXA-24-Ref', 'OXA-58-Ref') %>%
  rename('R number' = 'DCLS ID', 'Organism' = 'DCLS - ID confirmation', 
         'Receive Date' = 'Date received at DCLS') %>%
  unite('OXA-23', c('OXA-23','OXA-23-Ref'), na.rm = TRUE) %>%
  unite('OXA-24/40', c('OXA-24/40','OXA-24-Ref'), na.rm = TRUE) %>%
  unite('OXA-58', c('OXA-58','OXA-58-Ref'), na.rm = TRUE)

linelist_2023
total_2023 <- count(linelist_2023)
total_2023

#linelist_2022
names(linelist_2022) <- linelist_2022[1,]
linelist_2022 <- linelist_2022[-1,] # Rename column names as first row data
linelist_2022 <- linelist_2022 %>% #Selecting and renaming columns
  dplyr::select('DCLS ID', 'Submitter', 'OXA-23', 'OXA-24', 'OXA-58', 
                'Date of collection', 'DCLS - ID confirmation', 
                'Date received at DCLS', 'Source') %>%
  rename('R number' = 'DCLS ID', 'Organism' = 'DCLS - ID confirmation', 
         'Receive Date' = 'Date received at DCLS') %>%
  na.omit()
linelist_2022
total_2022 <- count(linelist_2022)
total_2022

#linelist_2021
names(linelist_2021) <- linelist_2021[1,]
linelist_2021 <- linelist_2021[-1,] # Rename column names as first row data
linelist_2021 <- linelist_2021 %>% #Selecting and renaming columns
  dplyr::select('DCLS ID', 'Submitter', 'Date of collection', 
                'DCLS - ID confirmation', 'Date received at DCLS', 
                'Source') %>%
  rename('R number' = 'DCLS ID', 'Organism' = 'DCLS - ID confirmation', 
         'Receive Date' = 'Date received at DCLS')%>%
  na.omit()
linelist_2021
total_2021 <- count(linelist_2021)
total_2021

#linelist_2020
names(linelist_2020) <- linelist_2020[1,]
linelist_2020 <- linelist_2020[-1,] # Rename column names as first row data
linelist_2020 <- linelist_2020 %>% #Selecting and renaming columns
  dplyr::select('DCLS ID', 'Submitter', 'Date of collection', 
                'DCLS - ID confirmation', 'Date received at DCLS', 
                'Source') %>%
  rename('R number' = 'DCLS ID', 'Organism' = 'DCLS - ID confirmation', 
         'Receive Date' = 'Date received at DCLS')
linelist_2020
total_2020 <- count(linelist_2020)
total_2020

linelist_2019
linelist_2019
linelist_2019 <- linelist_2019 %>% #Selecting and renaming columns
  dplyr::select('DCLS ID', 'Submitter', 'Date of collection', 
                'Submitter Organism ID', 'Date received at DCLS', 
                'Source') %>%
  rename('R number' = 'DCLS ID', 'Organism' = 'Submitter Organism ID', 
         'Receive Date' = 'Date received at DCLS') %>%
  filter(grepl('Acinetobacter|baum*', `Organism`))

total_2019 <- count(linelist_2019)
total_2019

#ncbi
ncbi <- ncbi %>% #Selecting columns
  dplyr::select(`Isolate identifiers`, `SNP cluster`, `AMR genotypes`) %>%
  rename('WGS ID#' = 'Isolate identifiers')
head(ncbi)

#WGS log
#select only A. baumannii sequences
wgs <- wgs %>%
  dplyr::select('R number', 'WGS ID#', `Organism genus`, `Organism species`) %>%
  unite(Organism, c(`Organism genus`, `Organism species`), sep = " ") %>%
  filter(grepl('baum*', `Organism`))
head(wgs)
summary(wgs)

#Editing R numbers to remove _# for rerun
#wgs
wgs$'R number' = gsub("_.*", "", wgs$'R number')

#Edit to remove duplicate R numbers from WGS data
wgs[!rev(duplicated(rev(wgs$'WGS ID#'))),]

#WGS log 2022
#select only A. baumannii sequences
#wgs2022 <- wgs2022 %>% #Selecting columns
#  dplyr::select('R number', 'WGS ID#', 'Organism') %>%
#  filter(grepl('bau*', `Organism`))
#head(wgs2022)
#summary(wgs2022)

#Editing R numbers to remove _# for rerun
#wgs2022
#wgs2022$'R number' = gsub("_.*", "", wgs2022$'R number')

#Edit to remove duplicate R numbers from WGS data
#wgs2022[!rev(duplicated(rev(wgs2022$'WGS ID#'))),]
```

```{r, include=FALSE}
#Combine all linelists into one dataframe
linelistall <- linelist_2020 %>%
  full_join(linelist_2021) %>%
  full_join(linelist_2023) %>%
  full_join(linelist_2022) %>%
  full_join(linelist_2019)
linelistall 
```

```{r, include=FALSE}
#removing 2024 data from 2023 report CHANGE EVERY YEAR
wgs
wgsall <- wgs[!grepl("2024EP*", wgs$`WGS ID#`),]
wgsall

#Fix typos in species names
wgsall$Organism = gsub(".*bau.*", "A. baumannii", wgsall$Organism)
wgsall
```

```{r, include=FALSE}

#merge the ncbi dataframe with the wgs log
wgsncbi <- merge(wgsall, ncbi, by = 'WGS ID#', all=TRUE)
wgsncbi
```

```{r, include=FALSE}
#merge the linelist dataframe with the ncbiwgs dataframe
alldata <- merge(wgsncbi, linelistall, by = 'R number', all=TRUE) 
alldata
```

```{r, include=FALSE}
#Cleaning Column data
#Merge Organism ID columns
alldata <- transform(alldata,
  Organism.x = replace(Organism.x, is.na(Organism.x), 
                       paste0(Organism.y[is.na(Organism.x)])))

#Remove 2nd Organism column after merging
alldata <- dplyr::select(alldata, -Organism.y)

#Rename Organism.x to Organism
alldata <- alldata %>% 
  rename(Organism = Organism.x)

#Replace NA in Collection date with receive date
alldata <- transform(alldata,
  Date.of.collection = replace(Date.of.collection, is.na(Date.of.collection), 
                               paste0(Receive.Date[is.na(Date.of.collection)])))

#Remove Receive date column after merging
alldata <- dplyr::select(alldata, -Receive.Date)
alldata
```

```{r, include=FALSE}
#Editing Submitter names
alldata$Submitter = gsub(".*Burlington.*", "LabCorp Burlington", alldata$Submitter)
alldata$Submitter = gsub(".*Mary Washington.*", "Mary Washington Hospital", alldata$Submitter)
alldata$Submitter = gsub(".*Quest.*", "Quest Diagnostics-Roanoke", alldata$Submitter)
alldata$Submitter = gsub(".*Sentara Norfolk.*", "Sentara Norfolk General Hospital", alldata$Submitter)
alldata$Submitter = gsub(".*Wythe.*", "Wythe County Community Hospital", alldata$Submitter)
alldata$Submitter = gsub(".*Henrico.*", "Henrico Doctor's Hospital", alldata$Submitter)
alldata$Submitter = gsub(".*Martha Jefferson.*", "Sentara Martha Jefferson Hospital", alldata$Submitter)
alldata$Submitter = gsub(".*PWMC.*", "Prince William Medical Center", alldata$Submitter)
alldata$Submitter = gsub(".*Prince William.*", "Prince William Medical Center", alldata$Submitter)
alldata$Submitter = gsub(".*Central VA.*", "McGuire VAMC", alldata$Submitter)
alldata$Submitter = gsub(".*Augusta.*", "Augusta Medical Center", alldata$Submitter)
alldata$Submitter = gsub(".*VA Medical Center - Hampton*", "Hampton VA Medical Center", alldata$Submitter)
alldata$Submitter = gsub(".*Winchester Medical  Center*", "Winchester Medical Center", alldata$Submitter)
```

```{r, include=FALSE}
#Add column of OXA genes identified using WGS
alldata <- alldata %>%  
  mutate(WGS.Gene = case_when(grepl("blaOXA-23.*blaOXA-72", AMR.genotypes) ~ "OXA-23, OXA-72",
                          grepl("OXA-23", AMR.genotypes) ~ "OXA-23",
                          grepl("OXA-24", AMR.genotypes) ~"OXA-24",
                          grepl("OXA-72", AMR.genotypes) ~"OXA-72", 
                          grepl("OXA-58", AMR.genotypes) ~"OXA-58",
                          grepl("OXA*,", AMR.genotypes) ~"Other OXA",
                          TRUE ~ "No OXA Genes"))

alldata
```

```{r,include=FALSE}
#Changing date of collection to date instead of character
alldata<- alldata %>% 
  mutate(Date.of.collection = as.Date(Date.of.collection, format = "%m/%d/%Y"))
alldata

#merge R numbers with multiple isolates sequenced
alldata <- alldata %>%
  fill(Submitter, Date.of.collection, Source, .direction = "down") %>%
  slice(n = c(-35, -53, -151, -155, -195))
alldata
```

```{r, include=FALSE}
#Add county of submitter
alldata <- alldata %>%  
   mutate(County = case_when(grepl("Augusta Medical Center", Submitter) ~ "Augusta County",
                           grepl("Chesapeake", Submitter) ~"Chesapeake city",
                           grepl("Prince William Medical Center", Submitter) ~"Prince William County", 
                           grepl("Labcorp Villa Park", Submitter) ~"Henrico County",
                           grepl("Mary Washington Hospital", Submitter) ~"Albemarle County",
                           grepl("Henrico Doctor's Hospital", Submitter) ~"Henrico County",
                           grepl("Quest Diagnostics-Roanoke", Submitter) ~"Roanoke city",
                           grepl("LabCorp Burlington", Submitter) ~"Alamance County",
                           grepl("CJW Medical Center", Submitter) ~"Richmond city",
                           grepl("Hazard ARH Reference Lab", Submitter) ~"Hazard County",
                           grepl("Sentara Martha Jefferson Hospital", Submitter) ~"Albemarle County",
                           grepl("Lynchburg General Hospital", Submitter) ~"Lynchburg city",
                           grepl("Winchester Medical Center", Submitter) ~"Frederick County",
                           grepl("McGuire VAMC", Submitter) ~"Richmond city",
                           grepl("Wythe County Community Hospital", Submitter) ~"Wythe County",
                           grepl("Riverside Regional Medical Center", Submitter) ~"Newport News city",
                           grepl("Children's Hospital of the King's Daughters", Submitter) ~"Norfolk city",
                           grepl("Virginia Hospital Center", Submitter) ~"Arlington County",
                           grepl("Sentara Norfolk General", Submitter) ~"Norfolk County",
                           grepl("Hampton VA Medical Center", Submitter) ~"Elizabeth City County",
                           grepl("Reston Hospital Center", Submitter) ~"Fairfax County",
                           grepl("Bon Secours HealthPartners Lab", Submitter) ~"Richmond city",
                           grepl("Richmond VA Medical Center", Submitter) ~"Richmond city",
                           grepl("Lewis Gale Medical Center", Submitter) ~"Roanoke County",
                           grepl("Central Regional Laboratories", Submitter) ~"Richmond city"))
alldata
```

```{r, include=FALSE}
#Add Health District
alldata <- alldata %>%  
   mutate("Health District" = case_when(grepl("Chesterfield", County) ~ "Chesterfield",
                           grepl("Colonial Heights", County) ~ "Chesterfield",
                           grepl("Powhatan", County) ~ "Chesterfield",
                           grepl("Dinwiddie", County) ~ "Crater",
                           grepl("Greensville", County) ~ "Crater",
                           grepl("Prince George", County) ~"Crater",
                           grepl("Surry", County) ~ "Crater",
                           grepl("Emporia", County) ~ "Crater",
                           grepl("Hopewell", County) ~ "Crater",
                           grepl("Petersburg", County) ~ "Crater",
                           grepl("Sussex", County) ~ "Crater",
                           grepl("Prince William", County) ~"Prince William",
                           grepl("Manassas city", County) ~"Prince William",
                           grepl("Manassas Park", County) ~"Prince William",
                           grepl("Henrico", County) ~"Henrico",
                           grepl("Charles City", County) ~"Chicahominy",
                           grepl("Goochland", County) ~"Chicahominy",
                           grepl("Hanover", County) ~"Chicahominy",
                           grepl("New Kent", County) ~"Chicahominy",
                           grepl("Amelia", County) ~"Piedmont",
                           grepl("Buckingham", County) ~"Piedmont",
                           grepl("Charlotte County", County) ~"Piedmont",
                           grepl("Cumberland", County) ~"Piedmont",
                           grepl("Lunenburg", County) ~"Piedmont",
                           grepl("Nottoway", County) ~"Piedmont",
                           grepl("Prince Edward", County) ~"Piedmont",
                           grepl("Roanoke city", County) ~"Roanoke",
                           grepl("Alamance", County) ~"North Carolina",
                           grepl("Hazard", Submitter) ~"Kentucky",
                           grepl("Richmond city", County) ~"Richmond",
                           grepl("Brunswick", County) ~"Southside",
                           grepl("Halifax", County) ~"Southside",
                           grepl("Mecklenburg", County) ~"Southside",
                           grepl("Chesapeake", County) ~"Chesapeake",
                           grepl("Accomack", County) ~"Eastern Shore",
                           grepl("Northampton", County) ~"Eastern Shore",
                           grepl("Hampton", County) ~"Hampton",
                           grepl("James City", County) ~"Peninsula",
                           grepl("Elizabeth City County", County) ~"Peninsula",
                           grepl("York", County) ~"Peninsula",
                           grepl("Newport News", County) ~"Peninsula",
                           grepl("Poquoson", County) ~"Peninsula",
                           grepl("Williamsburg", County) ~"Peninsula",
                           grepl("Portsmouth", County) ~"Portsmouth",
                           grepl("Norfolk", County) ~"Norfolk",
                           grepl("Essex", County) ~"Three Rivers",
                           grepl("Gloucester", County) ~"Three Rivers",
                           grepl("King and Queen", County) ~"Three Rivers",
                           grepl("King William", County) ~"Three Rivers",
                           grepl("Lancaster", County) ~"Three Rivers",
                           grepl("Mathews", County) ~"Three Rivers",
                           grepl("Middlesex", County) ~"Three Rivers",
                           grepl("Northumberland", County) ~"Three Rivers",
                           grepl("Richmond County", County) ~"Three Rivers",
                           grepl("Westmoreland", County) ~"Three Rivers",
                           grepl("Virginia Beach", County) ~"Virginia Beach",
                           grepl("Isle of Wight", County) ~"Western Tidewater",
                           grepl("Southampton", County) ~"Western Tidewater",
                           grepl("Franklin city", County) ~"Western Tidewater",
                           grepl("Suffolk", County) ~"Western Tidewater",
                           grepl("Alleghany", County) ~"Alleghany",
                           grepl("Botetourt", County) ~"Alleghany",
                           grepl("Craig", County) ~"Alleghany",
                           grepl("Covington", County) ~"Alleghany",
                           grepl("Salem", County) ~"Alleghany",
                           grepl("Roanoke County", County) ~"Alleghany",
                           grepl("Amherst", County) ~"Central Virginia",
                           grepl("Appomattox", County) ~"Central Virginia",
                           grepl("Bedford County", County) ~"Central Virginia",
                           grepl("Bedford city", County) ~"Central Virginia",
                           grepl("Campbell", County) ~"Central Virginia",
                           grepl("Lynchburg", County) ~"Central Virginia",
                           grepl("Amherst", County) ~"Central Virginia",
                           grepl("Buchanan", County) ~"Cumberland",
                           grepl("Dickenson", County) ~"Cumberland",
                           grepl("Russell", County) ~"Cumberland",
                           grepl("Tazewell", County) ~"Cumberland",
                           grepl("Lee", County) ~"Lenowisco",
                           grepl("Scott", County) ~"Lenowisco",
                           grepl("Wise", County) ~"Lenowisco",
                           grepl("Norton city", County) ~"Lenowisco",
                           grepl("Bland", County) ~"Mount Rogers",
                           grepl("Carroll", County) ~"Mount Rogers",
                           grepl("Grayson", County) ~"Mount Rogers",
                           grepl("Smyth", County) ~"Mount Rogers",
                           grepl("Washington", County) ~"Mount Rogers",
                           grepl("Wythe", County) ~"Mount Rogers",
                           grepl("Bristol", County) ~"Mount Rogers",
                           grepl("Galax", County) ~"Mount Rogers",
                           grepl("Floyd", County) ~"New River",
                           grepl("Giles", County) ~"New River",
                           grepl("Montgomery", County) ~"New River",
                           grepl("Pulaski", County) ~"New River",
                           grepl("Radford city", County) ~"New River",
                           grepl("Pittsylvania", County) ~"Pittsylvania-Danville",
                           grepl("Danville", County) ~"Pittsylvania-Danville",
                           grepl("Franklin County", County) ~"West Piedmont",
                           grepl("Henry", County) ~"West Piedmont",
                           grepl("Patrick", County) ~"West Piedmont",
                           grepl("Martinsville", County) ~"West Piedmont",
                           grepl("Alexandria", County) ~"Alexandria",
                           grepl("Fairfax city", County) ~"Fairfax",
                           grepl("Fairfax County", County) ~"Fairfax",
                           grepl("Falls Church", County) ~"Fairfax",
                           grepl("Loudoun", County) ~"Loudoun",
                           grepl("Augusta", County) ~"Central Shenandoah",
                           grepl("Bath", County) ~"Central Shenandoah",
                           grepl("Highland", County) ~"Central Shenandoah",
                           grepl("Rockbridge", County) ~"Central Shenandoah",
                           grepl("Rockingham", County) ~"Central Shenandoah",
                           grepl("Buena Vista", County) ~"Central Shenandoah",
                           grepl("Harrisonburg", County) ~"Central Shenandoah",
                           grepl("Lexington", County) ~"Central Shenandoah",
                           grepl("Staunton", County) ~"Central Shenandoah",
                           grepl("Waynesboro", County) ~"Central Shenandoah",
                           grepl("Clarke", County) ~"Lord Fairfax",
                           grepl("Frederick County", County) ~"Lord Fairfax",
                           grepl("Page", County) ~"Lord Fairfax",
                           grepl("Shenandoah", County) ~"Lord Fairfax",
                           grepl("Warren", County) ~"Lord Fairfax",
                           grepl("Winchester", County) ~"Lord Fairfax",
                           grepl("Caroline", County) ~"Rappahannock",
                           grepl("King George", County) ~"Rappahannock",
                           grepl("Spotsylvania", County) ~"Rappahannock",
                           grepl("Stafford", County) ~"Rappahannock",
                           grepl("Fredericksburg", County) ~"Rappahannock",
                           grepl("Culpeper", County) ~"Rappahannock/Rapidan",
                           grepl("Fauquier", County) ~"Rappahannock/Rapidan",
                           grepl("Madison", County) ~"Rappahannock/Rapidan",
                           grepl("Orange", County) ~"Rappahannock/Rapidan",
                           grepl("Rappahannock", County) ~"Rappahannock/Rapidan",
                           grepl("Albemarle", County) ~"Blue Ridge",
                           grepl("Fluvanna", County) ~"Blue Ridge",
                           grepl("Greene", County) ~"Blue Ridge",
                           grepl("Louisa", County) ~"Blue Ridge",
                           grepl("Nelson", County) ~"Blue Ridge",
                           grepl("Charlottesville", County) ~"Blue Ridge",
                           grepl("Arlington", County) ~"Arlington"))
alldata
```

```{r, include=FALSE}
# Remove NA in Cluster.ID to add info on whether not sequenced or not clustered
alldata <- alldata  %>%
  mutate(SNP.cluster = ifelse(is.na(WGS.ID.) & is.na(SNP.cluster), "Not Sequenced",
               ifelse(!is.na(WGS.ID.) & is.na(SNP.cluster), "No Cluster", SNP.cluster)))
alldata
```

```{r, include=FALSE}
#Add Latitude and Longitude values based on submitter
alldata <- alldata %>%  
   mutate(coordinates = case_when(grepl("Augusta Medical Center", Submitter) ~ "38.104478308423694, -78.94313755721967",
                        grepl("Chesapeake", Submitter) ~"36.74632545776263, -76.24545551359404",
                        grepl("Prince William Medical Center", Submitter) ~"38.76713259733973, -77.48812362559732", 
                        grepl("Labcorp Villa Park", Submitter) ~"37.632427232834395, -77.46820452324165",
                        grepl("Mary Washington Hospital", Submitter) ~"38.31011578121884, -77.48401184650754",
                        grepl("Henrico Doctor's Hospital", Submitter) ~"37.60414617832593, -77.53936392747299",
                        grepl("Quest Diagnostics-Roanoke", Submitter) ~"37.25137216267505, -79.93866182510648",
                        grepl("LabCorp Burlington", Submitter) ~"36.08328709818486, -79.51324203393507",
                        grepl("Hazard ARH Reference Lab", Submitter) ~"37.27829960068489, -83.22779160341811",
                        grepl("CJW Medical Center", Submitter) ~"37.51411512981057, -77.52649357933862",
                        grepl("Sentara Martha Jefferson Hospital", Submitter) ~"38.02322873567112, -78.4436675023191",
                        grepl("Lynchburg General Hospital", Submitter) ~"37.41669484982034, -79.17114611213256",
                        grepl("Winchester Medical Center", Submitter) ~"39.19432644925809, -78.19330350177302",
                        grepl("McGuire VAMC", Submitter) ~"37.498735642934136, -77.46752123911473",
                        grepl("Wythe County Community Hospital", Submitter) ~"36.954754124534425, -81.0964546251949",
                        grepl("Riverside Regional Medical Center", Submitter) ~"37.063284792870924, -76.48206272748374",
                        grepl("Children's Hospital of the King's Daughters", Submitter) ~"36.86275532199984, -76.30101499679927",
                        grepl("Virginia Hospital Center", Submitter) ~"38.88964244963296, -77.12711329675885",
                        grepl("Sentara Norfolk General", Submitter) ~"36.86297074298914, -76.30190302430464",
                        grepl("Hampton VA Medical Center", Submitter) ~"37.01551757229038, -76.33171476295321",
                        grepl("Reston Hospital Center", Submitter) ~"38.962484890795906, -77.36419895732466",
                        grepl("Bon Secours HealthPartners Lab", Submitter) ~"37.54136128646759, -77.39941152204372",
                        grepl("Richmond VA Medical Center", Submitter) ~"37.49684199691635, -77.46422205602433",
                        grepl("Lewis Gale Medical Center", Submitter) ~"37.26389679502726, -80.02823237291388",
                        grepl("Central Regional Laboratories", Submitter) ~"37.51411512981057, -77.52649357933862"))
alldata

#Separate coordinates into Latitude and Longitude
lat_long <- alldata %>%
  separate(coordinates, c('Latitude', 'Longitude'), ',')
lat_long$Latitude <- as.numeric(lat_long$Latitude)
lat_long$Longitude <- as.numeric(lat_long$Longitude)
lat_long
```

```{r, include=FALSE}
#Separate collection date into month, day, year
alldata_sep <- alldata %>%
  separate(Date.of.collection, c('Year', 'Month', 'Day'), '-')
alldata_sep

#Convert Month to numeric
alldata_sep$Month <- as.numeric(alldata_sep$Month)
alldata_sep

#Convert numerical month to name
alldata_sep$Month <- factor(month.abb[alldata_sep$Month],levels=month.abb)
alldata_sep

#Convert Year to numeric
alldata_sep$Year <- as.numeric(alldata_sep$Year)
alldata_sep
```
\noindent\rule{16cm}{0.4pt}
### CRAB = Carbapenem Resistant *Acinetobacter baumannii*
\noindent\rule{16cm}{0.4pt}

\noindent\rule{16cm}{0.4pt}
### Table of Contents
\noindent\rule{16cm}{0.4pt}
CRAB Submission.....................................................................................2  
blaOXA genes identified over time.....................................................................3  
blaOXA genes identified per submitter.................................................................4  
blaOXA genes per Virginia Health District............................................................6       
blaOXA gene frequency..............................................................................18  
CRAB clusters per NCBI.............................................................................23  
2020 data............................................................................................26  
2021 data............................................................................................28  
2022 data............................................................................................30  
2023 data............................................................................................32 

\noindent\rule{16cm}{0.4pt}
### Summary
\noindent\rule{16cm}{0.4pt}
CRAB submissions increased during 2020 when DCLS added CRAB testing.  During 2021 and 2022 the number of CRAB submissions leveled off and remained steady.  In 2023, the number of isolates sent for CRAB submission increased. The blaOXA genes identified using whole genome sequencing within the state of Virginia include OXA-23, OXA-24, and OXA-72 (an OXA-24-like gene).  For the purpose of this report these have been separated out to give an in depth look at levels of specific genes within Virginia.   OXA genes identified as other in this report possess either a gene from the OXA-51 family which are sometimes carbapenemase producing, or they have a novel OXA gene allele.  This report also includes mapped data showing which health districts and submitting facilities are sending *Acinetobacter baumannii* isolates and where genes are being identified.  However, these mapped data are based on submitting facility and may not be a true representation of where the cases are occurring due to the centralization of microbiology laboratory testing.  In addition, this report includes information on the clusters of isolates that were found to be related using DCLS genomic surveillance.  Several of these clusters carry over from year to year.  There is an overview of all the cluster data to show this carry over.  The cluster data and OXA gene data is also divided into subsections to show yearly data.

\newpage

Routine testing and whole genome sequencing was not performed on *Acinetobacter baumannii* until March 2020.   

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Histogram of CRAB submissions over time by month
plottitle <- expression(paste(italic("Acinetobacter baumannii "), 'Isolates Submitted per Month'))

alldata_year <- alldata %>% 
  mutate(Year = case_when(grepl("2019", Date.of.collection) ~ "2019",
                          grepl("2020", Date.of.collection) ~ "2020",
                          grepl("2021", Date.of.collection) ~ "2021",
                          grepl("2022", Date.of.collection) ~ "2022",
                          grepl("2023", Date.of.collection) ~ "2023"))

bar <- ggplot(data = alldata_year, aes(x = Date.of.collection, fill = Year)) +
  geom_histogram(color = "Black", bins = 30) +
  scale_fill_manual(values = c("#1F77B4", "#AEC7EB", "#FF7F0E", "#FFBB78", "#a5a7a8")) +
  scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("2 months")) +
  scale_y_continuous(expand = c(0,0)) +
  theme(  legend.position = "bottom",
        legend.key.size = unit(.5, 'cm'),
        legend.text=element_text(size=9), 
    legend.background = element_rect(fill="white",
                                  size=0.5, linetype="solid", 
                                  colour ="Black"), 
    legend.title = element_text(colour="black", size=10, 
                                      face="bold"),
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  xlab("Month of Collection") +
  ylab("Count") +
  labs(title = plottitle) +
  guides(x = guide_axis(angle = 45))
bar
```

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "Yearly CRAB Submissions and OXA genes Identified by Whole Genome Sequencing", tab.id = "tab 1", label = "tab 1"}

#Summarizing OXA gene data by year
Yearly_Submission2 <- alldata_sep %>%
  group_by(Year) %>%
  summarize('Isolates Submitted' = n(), 'Sequenced' = sum(!is.na(WGS.ID.)), 
            'OXA-23' = length(grep("OXA-23", AMR.genotypes)), 
            'OXA-24' = length(grep("OXA-24", AMR.genotypes)), 
            'OXA-72' =length(grep("OXA-72", AMR.genotypes)))

#Set table settings
set_flextable_defaults(
  font.family = "Roboto",
  font.size = 7, theme_fun = theme_vanilla)

#Table
table_1 <- flextable(Yearly_Submission2)
table_1 <- colformat_double(x = table_1,
  big.mark="", digits = 0)
table_1
```
Patients with multiple isolates submitted will only be sequencing once per year unless the PCR gene detected changes which is why more isolates were submitted than were sequenced.

\newpage

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#subset only sequenced isolates
sequenced <- subset(alldata, WGS.ID. != 'NA', na.rm = TRUE)
sequenced$WGS.Gene = gsub("OXA-23, OXA-72", "OXA-23 & OXA-72", sequenced$WGS.Gene)

#Create title with italics
plottitle1 <- expression(paste('OXA Genes in ', italic("Acinetobacter baumannii "),
                               'per Month'))
sequenced
#Bar graph of OXA genes per month
bar2 <- ggplot(data = sequenced) +
  geom_histogram(aes(x = Date.of.collection, group = WGS.Gene, fill = WGS.Gene),
                 color = "black", by = "months") +
  scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("2 months")) +
  scale_y_continuous(expand = c(0,0)) +
  guides(x = guide_axis(angle = 45)) +
  xlab("Month of Collection") + 
  ylab("Number of Isolates") +
  scale_fill_manual(values=OXA_Palette) +
  labs(title = plottitle1, caption = "Data for this graph includes only sequenced isolates") +
  guides(fill=guide_legend(title="Gene")) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  legend.position = "bottom",
        legend.key.size = unit(.5, 'cm'),
        legend.text=element_text(size=9), 
    legend.background = element_rect(fill="white",
                                  size=0.5, linetype="solid", 
                                  colour ="Black"), 
    legend.title = element_text(colour="black", size=10, 
                                      face="bold"))
bar2
```
OXA-72 is an OXA-24-like gene. For PCR testing both genes will result as OXA-24 positive.  From whole genome sequencing the gene alleles were detemined for the isolates and are divided by gene allele in this report to show the occurrence of each oxa-24-like gene allele in the state of Virginia.  Isolates identified as having other OXA genes possess genes from the OXA-51 family which are sometimes carbapenemase producers. There have been no cases of OXA-58 or OXA-235 identified in the state of Virginia to date.  Due to the inability to perform carbapenemase testing on *A. baumannii* isolates, DCLS sends all *A. baumannii* isolates with carbapenem antibiotic resistance observed on antimicrobial resistance susceptiblity testing for sequencing.  The isolates in the No OXA Genes present category did not have a carbapenemase gene predicted in their genome using whole genome sequencing.

\noindent\rule{16cm}{0.4pt}
### OXA Genes Per Submitter
\noindent\rule{16cm}{0.4pt}

The table and graphs below include data for all *A. baumannii* isolates submitted. If no OXA genes were identified this could be because the isolates were not sequenced or because there were no OXA genes identified by whole genome sequencing. 
\newpage
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "OXA Genes Per Submitter", tab.id = "tab 2", label = "tab 2"}

#Submitter and Gene Table
alldata %>% 
  select(Submitter, WGS.Gene) %>% 
  tbl_summary(by = WGS.Gene) %>%
  modify_header(label = "**Submitter**") %>% 
  add_overall() %>% 
  as_flex_table() %>% 
  flextable::fontsize(size = 7) %>% 
  fontsize(size = 7, part = "header") %>% 
  width(j = NULL, .8, unit = "in")
```

\newpage
\noindent\rule{16cm}{0.4pt}
### Per Health District Data
\noindent\rule{16cm}{0.4pt}

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Count OXA genes per health district
alldata_HD <-alldata %>% 
  count(`Health District`, WGS.Gene) 
alldata_HD$WGS.Gene = gsub("OXA-23, OXA-72", "OXA-23 & OXA-72", alldata_HD$WGS.Gene)
alldata
# bar graph of OXA genes per health district
ggplot(alldata_HD) +
  geom_col(mapping = aes(y = `Health District`, fill = WGS.Gene, x = n), color = "black") +
  xlab("Number of Isolates") +
  scale_fill_manual(values=OXA_Palette, name = "Gene") +
  scale_x_continuous(expand = c(0,0)) +
  labs(title = "OXA Genes per Health District", caption = "Graph based on all data") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  legend.position = "bottom",
        legend.key.size = unit(.5, 'cm'),
    legend.text=element_text(size=9), 
    legend.background = element_rect(fill="white",
                                  size=0.5, linetype="solid", 
                                  colour ="Black"), 
    legend.title = element_text(colour="black", size=10, 
                                      face="bold"))

```
Isolates from North Carolina and Kentucky were submitted by out of state laboratories but are Virginia residents. Isolate location is based off of submitting facility location not where the patient is located so this may not be a true representation of case location.

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all', fig.width=10, fig.height=11}
HD_count <- alldata %>% 
  count(`Health District`, WGS.Gene)
HD_count_sub <- subset(HD_count, WGS.Gene != 'No OXA Genes', na.rm = TRUE)
HD_count_sub$WGS.Gene = gsub("OXA-23, OXA-72", "OXA-23 & OXA-72", HD_count_sub$WGS.Gene)

#Bar graph of genes per county, Separated per Gene
ggplot(HD_count_sub) +
  geom_col(mapping = aes(y = `Health District`, fill = WGS.Gene, x = n), color = "black") +
  xlab("Number of Isolates") +
  scale_fill_manual(values=OXA_Palette, name ="Gene") +
  scale_x_continuous(expand = c(0,0)) +
  labs(title = "OXA Genes per Health District, Separated", caption = "Graph based on all data") +
 # gghighlight() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  legend.position = "none", 
           axis.text.x = element_text( size = 10),
           axis.title = element_text( size = 10, face = "bold" ),
           strip.text = element_text(size = 10)) +
  facet_wrap(~WGS.Gene)
```

Isolates from North Carolina and Kentucky were submitted by out of state laboratories but are Virginia residents. Isolate location is based off of the submitting facility's location not where the patient is located so this may not be a true representation of case location.

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
VA_map
VA_map1 <- VA_map %>% 
  arrange(NAME)
VA_map1
#Add Health District
HD_map <- VA_map %>%  
   mutate("Health District" = case_when(grepl("Chesterfield", NAMELSAD) ~ "Chesterfield",
                           grepl("Colonial Heights", NAMELSAD) ~ "Chesterfield",
                           grepl("Powhatan", NAMELSAD) ~ "Chesterfield",
                           grepl("Dinwiddie", NAMELSAD) ~ "Crater",
                           grepl("Greensville", NAMELSAD) ~ "Crater",
                           grepl("Prince George", NAMELSAD) ~"Crater",
                           grepl("Surry", NAMELSAD) ~ "Crater",
                           grepl("Emporia", NAMELSAD) ~ "Crater",
                           grepl("Hopewell", NAMELSAD) ~ "Crater",
                           grepl("Petersburg", NAMELSAD) ~ "Crater",
                           grepl("Sussex", NAMELSAD) ~ "Crater",
                           grepl("Prince William", NAMELSAD) ~"Prince William",
                           grepl("Manassas city", NAMELSAD) ~"Prince William",
                           grepl("Manassas Park", NAMELSAD) ~"Prince William",
                           grepl("Henrico", NAMELSAD) ~"Henrico",
                           grepl("Charles City", NAMELSAD) ~"Chicahominy",
                           grepl("Goochland", NAMELSAD) ~"Chicahominy",
                           grepl("Hanover", NAMELSAD) ~"Chicahominy",
                           grepl("New Kent", NAMELSAD) ~"Chicahominy",
                           grepl("Amelia", NAMELSAD) ~"Piedmont",
                           grepl("Buckingham", NAMELSAD) ~"Piedmont",
                           grepl("Charlotte County", NAMELSAD) ~"Piedmont",
                           grepl("Cumberland", NAMELSAD) ~"Piedmont",
                           grepl("Lunenburg", NAMELSAD) ~"Piedmont",
                           grepl("Nottoway", NAMELSAD) ~"Piedmont",
                           grepl("Prince Edward", NAMELSAD) ~"Piedmont",
                           grepl("Roanoke city", NAMELSAD) ~"Roanoke",
                           #grepl("Alamance", NAMELSAD) ~"North Carolina",
                           grepl("Richmond city", NAMELSAD) ~"Richmond",
                           grepl("Brunswick", NAMELSAD) ~"Southside",
                           grepl("Halifax", NAMELSAD) ~"Southside",
                           grepl("Mecklenburg", NAMELSAD) ~"Southside",
                           grepl("Chesapeake", NAMELSAD) ~"Chesapeake",
                           grepl("Accomack", NAMELSAD) ~"Eastern Shore",
                           grepl("Northampton", NAMELSAD) ~"Eastern Shore",
                           grepl("Hampton", NAMELSAD) ~"Hampton",
                           grepl("James City", NAMELSAD) ~"Peninsula",
                           grepl("York", NAMELSAD) ~"Peninsula",
                           grepl("Newport News", NAMELSAD) ~"Peninsula",
                           grepl("Poquoson", NAMELSAD) ~"Peninsula",
                           grepl("Williamsburg", NAMELSAD) ~"Peninsula",
                           grepl("Portsmouth", NAMELSAD) ~"Portsmouth",
                           grepl("Norfolk", NAMELSAD) ~"Norfolk",
                           grepl("Essex", NAMELSAD) ~"Three Rivers",
                           grepl("Gloucester", NAMELSAD) ~"Three Rivers",
                           grepl("King and Queen", NAMELSAD) ~"Three Rivers",
                           grepl("King William", NAMELSAD) ~"Three Rivers",
                           grepl("Lancaster", NAMELSAD) ~"Three Rivers",
                           grepl("Mathews", NAMELSAD) ~"Three Rivers",
                           grepl("Middlesex", NAMELSAD) ~"Three Rivers",
                           grepl("Northumberland", NAMELSAD) ~"Three Rivers",
                           grepl("Richmond County", NAMELSAD) ~"Three Rivers",
                           grepl("Westmoreland", NAMELSAD) ~"Three Rivers",
                           grepl("Virginia Beach", NAMELSAD) ~"Virginia Beach",
                           grepl("Isle of Wight", NAMELSAD) ~"Western Tidewater",
                           grepl("Southampton", NAMELSAD) ~"Western Tidewater",
                           grepl("Franklin city", NAMELSAD) ~"Western Tidewater",
                           grepl("Suffolk", NAMELSAD) ~"Western Tidewater",
                           grepl("Alleghany", NAMELSAD) ~"Alleghany",
                           grepl("Botetourt", NAMELSAD) ~"Alleghany",
                           grepl("Craig", NAMELSAD) ~"Alleghany",
                           grepl("Covington", NAMELSAD) ~"Alleghany",
                           grepl("Salem", NAMELSAD) ~"Alleghany",
                           grepl("Roanoke County", NAMELSAD) ~"Alleghany",
                           grepl("Amherst", NAMELSAD) ~"Central Virginia",
                           grepl("Appomattox", NAMELSAD) ~"Central Virginia",
                           grepl("Bedford County", NAMELSAD) ~"Central Virginia",
                           grepl("Bedford city", NAMELSAD) ~"Central Virginia",
                           grepl("Campbell", NAMELSAD) ~"Central Virginia",
                           grepl("Lynchburg", NAMELSAD) ~"Central Virginia",
                           grepl("Amherst", NAMELSAD) ~"Central Virginia",
                           grepl("Buchanan", NAMELSAD) ~"Cumberland",
                           grepl("Dickenson", NAMELSAD) ~"Cumberland",
                           grepl("Russell", NAMELSAD) ~"Cumberland",
                           grepl("Tazewell", NAMELSAD) ~"Cumberland",
                           grepl("Lee", NAMELSAD) ~"Lenowisco",
                           grepl("Scott", NAMELSAD) ~"Lenowisco",
                           grepl("Wise", NAMELSAD) ~"Lenowisco",
                           grepl("Norton", NAMELSAD) ~"Lenowisco",
                           grepl("Bland", NAMELSAD) ~"Mount Rogers",
                           grepl("Carroll", NAMELSAD) ~"Mount Rogers",
                           grepl("Grayson", NAMELSAD) ~"Mount Rogers",
                           grepl("Smyth", NAMELSAD) ~"Mount Rogers",
                           grepl("Washington", NAMELSAD) ~"Mount Rogers",
                           grepl("Wythe", NAMELSAD) ~"Mount Rogers",
                           grepl("Bristol", NAMELSAD) ~"Mount Rogers",
                           grepl("Galax", NAMELSAD) ~"Mount Rogers",
                           grepl("Floyd", NAMELSAD) ~"New River",
                           grepl("Giles", NAMELSAD) ~"New River",
                           grepl("Montgomery", NAMELSAD) ~"New River",
                           grepl("Pulaski", NAMELSAD) ~"New River",
                           grepl("Radford city", NAMELSAD) ~"New River",
                           grepl("Pittsylvania", NAMELSAD) ~"Pittsylvania-Danville",
                           grepl("Danville", NAMELSAD) ~"Pittsylvania-Danville",
                           grepl("Franklin County", NAMELSAD) ~"West Piedmont",
                           grepl("Henry", NAMELSAD) ~"West Piedmont",
                           grepl("Patrick", NAMELSAD) ~"West Piedmont",
                           grepl("Martinsville", NAMELSAD) ~"West Piedmont",
                           grepl("Alexandria", NAMELSAD) ~"Alexandria",
                           grepl("Fairfax city", NAMELSAD) ~"Fairfax",
                           grepl("Fairfax County", NAMELSAD) ~"Fairfax",
                           grepl("Falls Church", NAMELSAD) ~"Fairfax",
                           grepl("Loudoun", NAMELSAD) ~"Loudoun",
                           grepl("Augusta", NAMELSAD) ~"Central Shenandoah",
                           grepl("Bath", NAMELSAD) ~"Central Shenandoah",
                           grepl("Highland", NAMELSAD) ~"Central Shenandoah",
                           grepl("Rockbridge", NAMELSAD) ~"Central Shenandoah",
                           grepl("Rockingham", NAMELSAD) ~"Central Shenandoah",
                           grepl("Buena Vista", NAMELSAD) ~"Central Shenandoah",
                           grepl("Harrisonburg", NAMELSAD) ~"Central Shenandoah",
                           grepl("Lexington", NAMELSAD) ~"Central Shenandoah",
                           grepl("Staunton", NAMELSAD) ~"Central Shenandoah",
                           grepl("Waynesboro", NAMELSAD) ~"Central Shenandoah",
                           grepl("Clarke", NAMELSAD) ~"Lord Fairfax",
                           grepl("Frederick County", NAMELSAD) ~"Lord Fairfax",
                           grepl("Page", NAMELSAD) ~"Lord Fairfax",
                           grepl("Shenandoah", NAMELSAD) ~"Lord Fairfax",
                           grepl("Warren", NAMELSAD) ~"Lord Fairfax",
                           grepl("Winchester", NAMELSAD) ~"Lord Fairfax",
                           grepl("Caroline", NAMELSAD) ~"Rappahannock",
                           grepl("King George", NAMELSAD) ~"Rappahannock",
                           grepl("Spotsylvania", NAMELSAD) ~"Rappahannock",
                           grepl("Stafford", NAMELSAD) ~"Rappahannock",
                           grepl("Fredericksburg", NAMELSAD) ~"Rappahannock",
                           grepl("Culpeper", NAMELSAD) ~"Rappahannock/Rapidan",
                           grepl("Fauquier", NAMELSAD) ~"Rappahannock/Rapidan",
                           grepl("Madison", NAMELSAD) ~"Rappahannock/Rapidan",
                           grepl("Orange", NAMELSAD) ~"Rappahannock/Rapidan",
                           grepl("Rappahannock", NAMELSAD) ~"Rappahannock/Rapidan",
                           grepl("Albemarle", NAMELSAD) ~"Blue Ridge",
                           grepl("Fluvanna", NAMELSAD) ~"Blue Ridge",
                           grepl("Greene", NAMELSAD) ~"Blue Ridge",
                           grepl("Louisa", NAMELSAD) ~"Blue Ridge",
                           grepl("Nelson", NAMELSAD) ~"Blue Ridge",
                           grepl("Charlottesville", NAMELSAD) ~"Blue Ridge",
                           grepl("Arlington", NAMELSAD) ~"Arlington"))

HD_map1 <- HD_map %>% 
  dplyr::select(NAMELSAD, 'Health District') 
HD_map1
HD_map1 <- HD_map1 %>% 
  arrange(`Health District`)
HD_map1
#Turning off spherical geometry
sf_use_s2(FALSE)

#Grouping counties by Health District on the map
HD_group_map <- HD_map %>% 
  group_by(`Health District`) %>% 
  summarise() #%>% 
#  summarize(geometry = st_union(geometry)) 
HD_group_map
#Counting health district data by submission
HD_count <- alldata %>% 
  count(`Health District`) 
HD_count
#Rename columns
alldata_HD <- HD_count %>%
  rename("value" = "n")

#Joining Health Department data and map data
HD_data <- HD_group_map %>% 
  left_join(alldata_HD, by = "Health District") %>% 
  dplyr :: select("Health District", value, geometry) %>% 
  st_as_sf()  
HD_data
merged_HD <- fortify(HD_data, region = "Health District")
merged_HD

#buffering out county lines that don't match exactly
county_buff <- st_buffer(HD_data, .001)
HD_combined <- fortify(county_buff)

#Mapping health department data
map <-ggplot(merged_HD) + 
  geom_sf(aes(fill = value), color = NA) +
  scale_fill_gradientn(colours = c("#ffd7b5", "#ff6700"),
                       name="Submissions",
                       na.value = "lightgrey", 
                       values =  rescale(seq(0, 60))) +
  geom_sf(fill = "transparent", color = "gray20", size = .2, 
          data = HD_combined) + 
  theme(axis.line = element_blank(),
        axis.text = element_blank(),
        axis.ticks = element_blank(),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        legend.position = "bottom",
        legend.key.size = unit(.5, 'cm'),
        legend.text=element_text(size=9), 
        legend.background = element_rect(fill="white",
                                  size=0.5, linetype="solid", 
                                  colour ="Black"), 
        legend.title = element_text(colour="black", size=10, 
                                      face="bold")) +
  labs(title = "Submissions Per Health District") +
  coord_sf(xlim = c (-84, -75), ylim = c(35, 40))
map
```
Gray districts have no CRAB submissions. The data in this map is based off of submitter data. This is likely not a true representation of where these cases are occurring due to smaller facilities sending their microbiology testing to a larger central laboratory facility. In addition, submissions from North Carolina and Kentucky reference laboratories are also not included in these maps.

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Counting County data by submission
Sub_county <- alldata %>% 
  count(County) %>% 
  rename("Submissions" = "n")

#Rename columns
alldata_counties <- Sub_county %>%
  rename("region" = "County", "value" = "Submissions")

#join the county data to the VA_map
County_data <- VA_map %>% 
  left_join(alldata_counties, by = c("NAMELSAD" = "region")) %>% 
  dplyr :: select(NAMELSAD, value, geometry) %>% 
  st_as_sf()  

#choropleth map of county data
ggplot(County_data) + 
  geom_sf(aes(fill = value)) +
  scale_fill_gradientn(colours = c("dark Blue", "Green"),
                       name="Submissions",
                       na.value = "lightgrey", 
                       values =  rescale(seq(0, 200))) +
  labs(title = "Submissions Per County") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(),
  axis.line = element_blank(),
  axis.text = element_blank(),
  axis.ticks = element_blank(),
  legend.position = "bottom",
        legend.key.size = unit(.5, 'cm'),
    legend.text=element_text(size=9), 
    legend.background = element_rect(fill="white",
                                  size=0.5, linetype="solid", 
                                  colour ="Black"), 
    legend.title = element_text(colour="black", size=10, 
                                      face="bold")) +
  coord_sf(xlim = c (-84, -75), ylim = c(35, 40))

```
Gray counties have no CRAB submissions.  The data in this map is based off of submitter data. This is likely not a true representation of where these cases are occurring due to smaller facilities sending their microbiology testing to a larger central laboratory facility. In addition, submissions from North Carolina and Kentucky reference laboratories are also not included in these maps.

\newpage

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "Submitters Per Health District", tab.id = "tab 3", label = "tab 3"}
#Grouping and Summarizing submission data per health district
HD_Submitters <- alldata %>%
  select(`Health District`, Submitter) %>%
  group_by(`Health District`, Submitter) %>%
  summarize('Isolates' = n()) %>%
  ungroup()

#To group by Health District
HD_Summary <- as_grouped_data(x = HD_Submitters, groups = c("Health District"), columns = NULL)

#Submitters in Health District Table
table3 <- flextable(HD_Summary) 
table3 <- autofit(table3)
table3

```
Isolates from North Carolina and Kentucky were submitted by out of state laboratories but are Virginia residents.  Isolate location is based off of the submitting facility's location not where the patient is located so this may not be a true representation of case location.

\newpage

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Subset data for OXA-23
OXA_23 <- subset(alldata, WGS.Gene == 'OXA-23'|WGS.Gene == 'OXA-23, OXA-72', na.rm = TRUE)

#Counting health district data by OXA-23 +
HD_oxa23_count <- OXA_23 %>% 
  count(`Health District`)

#Rename columns
alldata_HD2 <- HD_oxa23_count %>%
  rename("region" = "Health District", "value" = "n")

HD_group_map

#Joining Health Department OXA-23 data and map data
HD_data2 <- HD_group_map %>% 
  left_join(alldata_HD2, by = c("Health District" = "region")) %>% 
  dplyr :: select("Health District", value, geometry) %>% 
  st_as_sf()  

merged_HD <- fortify(HD_data2, region = "Health District")

#buffering out county lines that don't match exactly
county_buff <- st_buffer(HD_data2, .001)
HD_combined1 <- fortify(county_buff)

#Mapping health department data
ggplot(merged_HD) + 
  geom_sf(aes(fill = value), color = NA) +
  scale_fill_gradientn(colours = c("Yellow", "Red"),
                       name="Isolates",
                       na.value = "lightgrey", 
                       values =  rescale(seq(0, 20))) +
  geom_sf(fill = "transparent", color = "gray20", size = .2, 
          data = HD_combined1) + 
  labs(title = "OXA-23 Per Health District") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), 
  axis.line = element_blank(),
  axis.text = element_blank(),
  axis.ticks = element_blank(),
  legend.position = "bottom",
        legend.key.size = unit(.5, 'cm'),
    legend.text=element_text(size=9), 
    legend.background = element_rect(fill="white",
                                  size=0.5, linetype="solid", 
                                  colour ="Black"), 
    legend.title = element_text(colour="black", size=10, 
                                      face="bold")) +
  coord_sf(xlim = c (-84, -75), ylim = c(35, 40))

```
Gray districts have no OXA-23 genes identified. The data in this map is based off of submitter data. This is likely not a true representation of where these cases are occurring due to smaller facilities sending their microbiology testing to a larger central laboratory facility. In addition, submissions from North Carolina and Kentucky reference laboratories are also not included in these maps.


\newpage
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "OXA-23 Positive Isolates Per Health District", tab.id = "tab 4", label = "tab 4"}

#table of data
table_4 <- flextable(HD_oxa23_count) %>%
  set_header_labels(n = "Number of Isolates")

table_4
```
Isolates from North Carolina and Kentucky were submitted by out of state laboratories but are Virginia residents.  Isolate location is based off of the submitting facility's location not where the patient is located so this may not be a true representation of case location.

\newpage

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Subset data for OXA-24
OXA_24 <- subset(alldata, WGS.Gene == 'OXA-24', na.rm = TRUE)
OXA_24

#Counting health district data by OXA-24 +
HD_oxa24_count <- OXA_24 %>% 
  count(`Health District`)

#Rename columns
alldata_HD2 <- HD_oxa24_count %>%
  rename("region" = "Health District", "value" = "n")
alldata_HD2
HD_group_map

#Joining Health Department OXA-24 data and map data
HD_data2 <- HD_group_map %>% 
  left_join(alldata_HD2, by = c("Health District" = "region")) %>% 
  dplyr :: select("Health District", value, geometry) %>% 
  st_as_sf()  
HD_data2
merged_HD <- fortify(HD_data2, region = "Health District")

#buffering out county lines that don't match exactly
county_buff <- st_buffer(merged_HD, .001)
HD_combined1 <- fortify(county_buff)

#Mapping health department data
ggplot(merged_HD) + 
  geom_sf(aes(fill = value), color = NA) +
  scale_fill_gradientn(colours = c("Light Blue", "dark Blue"),
                       name="Isolates",
                       na.value = "lightgrey", 
                       values =  rescale(seq(0, 20))) +
  geom_sf(fill = "transparent", color = "gray20", size = .2, 
          data = HD_combined1) + 
  labs(title = "OXA-24 Per Health District") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), 
  axis.line = element_blank(),
  axis.text = element_blank(),
  axis.ticks = element_blank(),
  legend.position = "bottom",
        legend.key.size = unit(.5, 'cm'),
    legend.text=element_text(size=9), 
    legend.background = element_rect(fill="white",
                                  size=0.5, linetype="solid", 
                                  colour ="Black"), 
    legend.title = element_text(colour="black", size=10, 
                                      face="bold")) +
  coord_sf(xlim = c (-84, -75), ylim = c(35, 40))
```
Gray districts have no OXA-24 genes identified.  The data in this map is based off of submitter data. This is likely not a true representation of where these cases are occurring due to smaller facilities sending their microbiology testing to a larger central laboratory facility. In addition, submissions from North Carolina and Kentucky reference laboratories are also not included in these maps.


\newpage
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "OXA-24 Positive Isolates Per Health District", tab.id = "tab 5", label = "tab 5"}

#table of data
table_5 <- flextable(HD_oxa24_count)%>%
  set_header_labels(n = "Number of Isolates")
table_5
```
Isolates from North Carolina and Kentucky were submitted by out of state laboratories but are Virginia residents.  Isolate location is based off of the submitting facility's location not where the patient is located so this may not be a true representation of case location.

\newpage

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Subset data for OXA-72
OXA_72 <- subset(alldata, WGS.Gene == 'OXA-72'|WGS.Gene == 'OXA-23, OXA-72', na.rm = TRUE)

#Counting health district data by OXA-72 +
HD_oxa_count <- OXA_72 %>% 
  count(`Health District`)

#Rename columns
alldata_HD2 <- HD_oxa_count %>%
  rename("region" = "Health District", "value" = "n")
alldata_HD2

#Joining Health Department OXA-72 data and map data
HD_data2 <- HD_group_map %>% 
  left_join(alldata_HD2, by = c("Health District" = "region")) %>% 
  dplyr :: select("Health District", value, geometry) %>% 
  st_as_sf()  
HD_data2
merged_HD <- fortify(HD_data2, region = "Health District")

#buffering out county lines that don't match exactly
county_buff <- st_buffer(merged_HD, .001)
HD_combined1 <- fortify(county_buff)

#Mapping health department data
ggplot(merged_HD) + 
  geom_sf(aes(fill = value), color = NA) +
  scale_fill_gradientn(colours = c("light green", "Dark Green"),
                       name="Isolates",
                       na.value = "lightgrey", 
                       values =  rescale(seq(0, 20))) +
  geom_sf(fill = "transparent", color = "gray20", size = .2, 
          data = HD_combined1) +
  labs(title = "OXA-72 Per Health District") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), 
  axis.line = element_blank(),
  axis.text = element_blank(),
  axis.ticks = element_blank(),
  legend.position = "bottom",
        legend.key.size = unit(.5, 'cm'),
    legend.text=element_text(size=9), 
    legend.background = element_rect(fill="white",
                                  size=0.5, linetype="solid", 
                                  colour ="Black"), 
    legend.title = element_text(colour="black", size=10, 
                                      face="bold")) +
  coord_sf(xlim = c (-84, -75), ylim = c(35, 40))
```
Gray districts have no OXA-72 genes identified.  The data in this map is based off of submitter data. This is likely not a true representation of where these cases are occurring due to smaller facilities sending their microbiology testing to a larger central laboratory facility. In addition, submissions from North Carolina and Kentucky reference laboratories are also not included in these maps.


\newpage
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "OXA-72 Positive Isolates Per Health District", tab.id = "tab 6", label = "tab 6"}
#table of data
table_6 <- flextable(HD_oxa_count)%>%
  set_header_labels(n = "Number of Isolates")
table_6
```
Isolates from North Carolina and Kentucky were submitted by out of state laboratories but are Virginia residents.  Isolate location is based off of the submitting facility's location not where the patient is located so this may not be a true representation of case location.

\newpage
\noindent\rule{16cm}{0.4pt}
### OXA Gene Frequency
\noindent\rule{16cm}{0.4pt}

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#selecting sequenced isolates from all the data
OXA_Genes <- subset(alldata, WGS.ID. != 'NA', na.rm = TRUE)
OXA_Genes <- subset(OXA_Genes, WGS.Gene != 'No OXA Genes', na.rm = TRUE)
OXA23_Genes <- subset(OXA_Genes, WGS.Gene == 'OXA-23', na.rm = TRUE)
OXA24_Genes <- subset(OXA_Genes, WGS.Gene == 'OXA-24', na.rm = TRUE)
OXA72_Genes <- subset(OXA_Genes, WGS.Gene == 'OXA-72', na.rm = TRUE)
OXA2372_Genes <- subset(OXA_Genes, WGS.Gene == 'OXA-23, OXA-72', na.rm = TRUE)
OtherOXA_Genes <- subset(OXA_Genes, WGS.Gene == 'Other OXA', na.rm = TRUE)

#counting genes IDed by WGS
OXA_Genes <- OXA_Genes %>%
  group_by(WGS.Gene, Date.of.collection) %>%
  summarize('Isolates' = n())
#OXAmedian <- OXA_Genes %>%
#    group_by(WGS.Gene) %>%
#    summarise(mediany = median(Isolates))


#OXAmean <- OXA_Genes %>%
#    group_by(WGS.Gene) %>%
#    summarise(meanY = mean(Isolates))

#creating italicized title for freq plot
plottitle2 <- expression(paste('Frequency of OXA-23 Gene in ', 
                               italic("Acinetobacter baumannii")))
plottitle3 <- expression(paste('Frequency of OXA-24 Gene in ', 
                               italic("Acinetobacter baumannii")))
plottitle4 <- expression(paste('Frequency of OXA-72 Gene in ', 
                               italic("Acinetobacter baumannii")))
plottitle5 <- expression(paste('Frequency of both OXA-23 & OXA-72 Genes in the same isolate of ', 
                               italic("Acinetobacter baumannii")))
plottitle6 <- expression(paste('Frequency of Other OXA Genes in ', 
                               italic("Acinetobacter baumannii")))

#OXA-23 Frequency
ggplot(data = OXA23_Genes, mapping = aes(x = Date.of.collection)) +
  geom_freqpoly(binwidth = 30, size = 1, color = "#1F77B4") +
    scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("2 months")) +
  scale_y_continuous(breaks = pretty_breaks(), expand = c(0,0.05)) +
  labs(title=expression()) +
  theme_bw() +
  xlab("Month of Collection") +
  ylab("Number of Isolates") +
  labs(title = plottitle2) +
  guides(x = guide_axis(angle = 45))+  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  plot.title = element_text(size=10),
  legend.position = "none", 
           axis.text.x = element_text( size = 10),
           axis.title = element_text( size = 10, face = "bold" ),
           strip.text = element_text(size = 10)) 

#OXA-24 Frequency
ggplot(data = OXA24_Genes, mapping = aes(x = Date.of.collection)) +
  geom_freqpoly(binwidth = 30, size = 1, color = "#FF7F0E") +
    scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("2 months")) +
  scale_y_continuous(breaks = pretty_breaks(), expand = c(0,0.05)) +
  labs(title=expression()) +
  theme_bw() +
  xlab("Month of Collection") +
  ylab("Number of Isolates") +
  labs(title = plottitle3) +
  guides(x = guide_axis(angle = 45))+  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  plot.title = element_text(size=10),
  legend.position = "none", 
           axis.text.x = element_text( size = 10),
           axis.title = element_text( size = 10, face = "bold" ),
           strip.text = element_text(size = 10)) 

#OXA-72 Frequency
ggplot(data = OXA72_Genes, mapping = aes(x = Date.of.collection)) +
  geom_freqpoly(binwidth = 30, size = 1, color = "#FFBB78") +
    scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("2 months")) +
  scale_y_continuous(breaks = pretty_breaks(), expand = c(0,0.05)) +
  labs(title=expression()) +
  theme_bw() +
  xlab("Month of Collection") +
  ylab("Number of Isolates") +
  labs(title = plottitle4) +
  guides(x = guide_axis(angle = 45))+  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  plot.title = element_text(size=10),
  legend.position = "none", 
           axis.text.x = element_text( size = 10),
           axis.title = element_text( size = 10, face = "bold" ),
           strip.text = element_text(size = 10)) 

#OXA-23&72 Frequency
ggplot(data = OXA2372_Genes, mapping = aes(x = Date.of.collection)) +
  geom_freqpoly(binwidth = 30, size = 1, color = "#AEC7EB") +
    scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("1 months")) +
  scale_y_continuous(breaks = c(1, 2, 3), expand = c(0,0.05)) +
  labs(title=expression()) +
  theme_bw() +
  xlab("Month of Collection") +
  ylab("Number of Isolates") +
  labs(title = plottitle5) +
  guides(x = guide_axis(angle = 45))+  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  plot.title = element_text(size=10),
  legend.position = "none", 
           axis.text.x = element_text( size = 10),
           axis.title = element_text( size = 10, face = "bold" ),
           strip.text = element_text(size = 10)) 

#Other OXA Frequency
ggplot(data = OtherOXA_Genes, mapping = aes(x = Date.of.collection)) +
  geom_freqpoly(binwidth = 30, size = 1, color = "#2CA02C") +
  scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("2 months")) +
  scale_y_continuous(breaks = c(1, 2, 3), expand = c(0,0.05)) + #May need to adjust breaks annually
  labs(title=expression()) +
  theme_bw() +
  xlab("Month of Collection") +
  ylab("Number of Isolates") +
  labs(title = plottitle6) +
  guides(x = guide_axis(angle = 45))+  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  plot.title = element_text(size=10),
  legend.position = "none", 
           axis.text.x = element_text( size = 10),
           axis.title = element_text( size = 10, face = "bold" ),
           strip.text = element_text(size = 10)) 

```

\newpage
\noindent\rule{16cm}{0.4pt}
## *A. baumannii* Clusters Per NCBI
\noindent\rule{16cm}{0.4pt}
NCBI is the National Center for Biotechnology Information (https://www.ncbi.nlm.nih.gov/).  Public health laboratories receiving CDC funding for carbapenem-resistant organism sequencing are required to submit sequences to the NCBI Pathogen Detection database.  One advantage of submission to Pathogen Detection is for broad swath surveillance for potential genetically related isolates among all isolate sequences submitted to NCBI under organism-specific umbrella BioProjects surveilled by Pathogen Detection. Adopting NGS methods for detection of clusters of AMR bacterial isolates that are genetically similar using a single nucleotide polymorphism (SNP) comparison method enables genomic surveillance of CRAB isolates. Clusters identified by NCBI are then verified by DCLS and assigned DCLS cluster ID codes and reported to VDH district epidemiologists. 
Isolates from North Carolina and Kentucky were submitted by out of state laboratories but are Virginia residents. Any clusters with less than 3 Virginia isolates have been removed from this graph.  Isolate location is based off of the submitting facility's location not where the patient is located so this may not be a true representation of case location.

\newpage

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all', fig.width=10,fig.height=13}
#Renaming NCBI clusters
Cluster_only2 <- subset(alldata, SNP.cluster!= 'Not Sequenced')
Cluster_only2 <- subset(Cluster_only2, SNP.cluster != 'No Cluster')
Cluster_only2 <- Cluster_only2 %>%  
   mutate("Cluster.ID" = case_when(grepl("PDS000170123.12", SNP.cluster) ~ ("VA_AB_OXA24_220914 \n or VA_AB_OXA24_220120"),
                                   grepl("PDS000075026.35", SNP.cluster) ~ "VA_AB_OXA23_240205",
                                   grepl("PDS000163224.5", SNP.cluster) ~ "VA_AB_OXA23OXA24_231117",
                                   grepl("PDS000167669.19", SNP.cluster) ~ "VA_AB_OXA23_230424",
                                   grepl("PDS000084878.70", SNP.cluster) ~ "VA_AB_OXA24_230526",
                                   grepl("PDS000098000.39", SNP.cluster) ~ "VA_AB_OXA24_231024",
                                   grepl("PDS000075027.39", SNP.cluster) ~ "VA_AB_OXA24_231119",
                                   grepl("PDS000047378.30", SNP.cluster) ~ "VA_AB_OXA23_220720", 
                                   grepl("PDS000131686.49", SNP.cluster) ~ "VA_AB_OXA23_230626",
                                   grepl("PDS000167669.19", SNP.cluster) ~ "VA_AB_OXA23_230424",
                                   grepl("PDS000107408.24", SNP.cluster) ~ "VA_AB_OXA23_230125",
                                   grepl("PDS000172394.2", SNP.cluster) ~ "VA_AB_OXA23NDM_240106",
                                   grepl("PDS000120625.5", SNP.cluster) ~ "VA_AB_OXA23_221103",
                                   grepl("PDS000133955.7", SNP.cluster) ~ "VA_AB_OXA23_230424-2",
                                   grepl("PDS000104873.4", SNP.cluster) ~ "Duplicate patient",
                                   grepl("PDS000107407.10", SNP.cluster) ~ "Duplicate patient",
                                   grepl("PDS000112545.7", SNP.cluster) ~ "SNP >10",
                                   grepl("PDS000122101.4", SNP.cluster) ~ "SNP >10",
                                   grepl("PDS000148754.51", SNP.cluster) ~ "SNP >10",
                                   grepl("PDS000170122.3", SNP.cluster) ~ "SNP >10",
                                   grepl("PDS000150588.3", SNP.cluster) ~ "SNP >10"))

#Rename Cluster ID with NCBI Cluster name if Cluster ID is NA
Cluster_only2$Cluster.ID <- ifelse(is.na(Cluster_only2$Cluster.ID), 
                       yes = Cluster_only2$SNP.cluster,
                       no = Cluster_only2$Cluster.ID)

Cluster_only2 <- subset(Cluster_only2, Cluster.ID!= 'SNP >10')
Cluster_only2 <- subset(Cluster_only2, Cluster.ID!= 'Duplicate patient')


#Grouping and Summarizing the data per health district
Clusters3 <- Cluster_only2 %>%
  group_by(`Cluster.ID`, `Health District`, Date.of.collection) %>%
  summarize('Isolates' = n()) %>%
  ungroup() %>%
  group_by (`Cluster.ID`) %>%
  mutate(total_isolates = sum(Isolates)) %>%
  filter(total_isolates >= 3)

#bar graph of clusters per county over time
#ggplot(Clusters3) +
#  geom_histogram(aes(x=Date.of.collection, fill = `Health District`), by = "months",
#                 color = "Black", binwidth= 31) +
#  scale_fill_manual(values=Palette) +
#  facet_wrap(reformulate("Cluster.ID", "."), scales = "free_x", nrow = 4) +
#  xlab("Month of Collection") +
  # ylab("Number of Isolates") +
  # scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("2 months")) + 
  # theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  # panel.background = element_blank(), axis.line = element_line(colour = "black"),
  # axis.text.x = element_text( size = 8),
  #       axis.text.y = element_text( size = 8),
  #       axis.title = element_text(size = 8, face = "bold", hjust = 0.5),
  #       legend.position = "bottom",
  #       legend.key.size = unit(.2, 'cm'),
  #       legend.text=element_text(size=8),
  #       strip.text.y = element_text(angle = 0, size = 8)) +
#   theme(panel.grid.minor = element_blank())+ #Remove grid lines for non-break numbers
 
  # labs(title = "Cluster Isolates per Health District")+
  # guides(x = guide_axis(angle = 45))

#bar graph of clusters per county over time
ggplot(Clusters3) +
  geom_histogram(aes(x=Date.of.collection, fill = `Health District`), by = "months",
                 color = "Black", binwidth= 31) +
  scale_fill_manual(values=Palette) +
  facet_grid(reformulate(".", "Cluster.ID"), scales = "free_x") +
  xlab("Month of Collection") +
  ylab("Number of Isolates") +
  scale_x_date(labels=date_format ("%b %y"), breaks=date_breaks("2 months")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  axis.text.x = element_text( size = 10),
        axis.text.y = element_text( size = 10),
        axis.title = element_text(size = 10, face = "bold", hjust = 0.5),
        legend.position = "bottom",
        legend.key.size = unit(.2, 'cm'),
        legend.text=element_text(size=10),
        strip.text.y = element_text(angle = 0, size = 10)) +
#   theme(panel.grid.minor = element_blank())+ #Remove grid lines for non-break numbers
  labs(title = "Cluster Isolates per Health District")+
  guides(x = guide_axis(angle = 45), fill = guide_legend(title.position = "top", title.hjust=0.5))
```


\newpage
Table 7: Isolates per DCLS Cluster ID by Year
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "Isolates per DCLS Cluster ID by Year", tab.id = "tab 7", label = "tab 7"}
ClustersGene <- Clusters3 %>%
  separate(Date.of.collection, c('Year', 'Month', 'Day'), '-')


#Changes the Column names to be table ready
colnames(ClustersGene)[colnames(ClustersGene) == 'Cluster.ID'] <- 'DCLS Cluster ID'

#Creates Genes per cluster table (simplified version)
table_7 <- ClustersGene %>% 
  select('DCLS Cluster ID', Year) %>%
  tbl_cross(row = 'DCLS Cluster ID',
            col = Year,
            #percent = "row",
            statistic = NULL)%>%
  bold_labels()
table_7
```

\newpage
\noindent\rule{16cm}{0.4pt}
### 2020
\noindent\rule{16cm}{0.4pt}

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Divide and count CRAB sequencing total by year 2020
alldata_sep <- alldata_sep %>%  
   mutate("Cluster.ID" = case_when(grepl("PDS000170123.12", SNP.cluster) ~ "VA_AB_OXA24_220914 \n or VA_AB_OXA24_220120",
                                   grepl("PDS000075026.35", SNP.cluster) ~ "VA_AB_OXA23_240205",
                                   grepl("PDS000163224.5", SNP.cluster) ~ "VA_AB_OXA23OXA24_231117",
                                   grepl("PDS000167669.19", SNP.cluster) ~ "VA_AB_OXA23_230424",
                                   grepl("PDS000084878.70", SNP.cluster) ~ "VA_AB_OXA24_230526",
                                   grepl("PDS000098000.39", SNP.cluster) ~ "VA_AB_OXA24_231024",
                                   grepl("PDS000075027.39", SNP.cluster) ~ "VA_AB_OXA24_231119",
                                   grepl("PDS000047378.30", SNP.cluster) ~ "VA_AB_OXA23_220720", 
                                   grepl("PDS000131686.49", SNP.cluster) ~ "VA_AB_OXA23_230626",
                                   grepl("PDS000167669.19", SNP.cluster) ~ "VA_AB_OXA23_230424",
                                   grepl("PDS000107408.24", SNP.cluster) ~ "VA_AB_OXA23_230125",
                                   grepl("PDS000172394.2", SNP.cluster) ~ "VA_AB_OXA23NDM_240106",
                                   grepl("PDS000120625.5", SNP.cluster) ~ "VA_AB_OXA23_221103",
                                   grepl("PDS000133955.7", SNP.cluster) ~ "VA_AB_OXA23_230424-2",
                                   grepl("PDS000104873.4", SNP.cluster) ~ "Duplicate patient",
                                   grepl("PDS000107407.10", SNP.cluster) ~ "Duplicate patient",
                                   grepl("PDS000112545.7", SNP.cluster) ~ "SNP >10",
                                   grepl("PDS000122101.4", SNP.cluster) ~ "SNP >10",
                                   grepl("PDS000148754.51", SNP.cluster) ~ "SNP >10",
                                   grepl("PDS000170122.3", SNP.cluster) ~ "SNP >10",
                                   grepl("PDS000150588.3", SNP.cluster) ~ "SNP >10"))

alldata_sep <- subset(alldata_sep, Cluster.ID!= 'SNP >10')
alldata_sep <- subset(alldata_sep, Cluster.ID!= 'Duplicate patient')
alldata_sep$WGS.Gene = gsub("OXA-23, OXA-72", "OXA-23 & OXA-72", alldata_sep$WGS.Gene)

crab_2020 <- alldata_sep %>%
  filter(Year == 2020)

#Selecting only 2020 data and Graphing Clusters
crab_2020 %>% 
  count(Submitter, Cluster.ID) %>% 
  ggplot() +
  geom_col(mapping = aes(y = Submitter, fill = factor(Cluster.ID, levels=c("VA_AB_OXA23OXA24_231117","VA_AB_OXA23_230424", "VA_AB_OXA24_230526", "VA_AB_OXA24_231024", "VA_AB_OXA23_220720", "VA_AB_OXA23_230626", "VA_AB_OXA23_230424-1",  "VA_AB_OXA23NDM_240106", "VA_AB_OXA23_221103", "VA_AB_OXA23_230424-2", "VA_AB_OXA23_240205", "VA_AB_OXA23_230125", "VA_AB_OXA24_231119", "VA_AB_OXA24_220914 \n or VA_AB_OXA24_220120")), x = n), 
           color = "black") +
  scale_fill_manual(values=Cluster_Palette) +
  scale_x_continuous(expand = c(0,0)) +
  xlab("Number of Isolates") +
  labs(title = "2020 Clusters per Submitter") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  legend.key.size = unit(.5, 'cm'),
        legend.text=element_text(size=9)) +
  guides(fill=guide_legend(title="DCLS Cluster ID")) #Rename legend
```
DCLS Cluster IDs have been color coded to assist in identifying clusters from one year to the next in this and the following graphs of yearly data.

\newpage
Table 8: 2020 OXA Genes Identified Per Cluster
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "2020 OXA Genes Identified Per Cluster", tab.id = "tab 8", label = "tab 8"}

#Genes per cluster table (old table too complicated)
#crab_2020 %>% 
#  select(Cluster.ID, WGS.Gene) %>% 
#  tbl_summary(by = WGS.Gene) %>%
#  modify_header(label = "**Cluster ID**") %>% 
#  add_overall() %>% 
#  as_flex_table() %>% 
#  flextable::fontsize(size = 7) %>% 
#  fontsize(size = 7, part = "header") %>% 
#  width(j = NULL, .85, unit = "in") %>%
#  flextable::delete_rows(i = 1, part = "body") %>%
#  autofit()

#Changes the Column names to be table ready
colnames(crab_2020)[colnames(crab_2020) == 'Cluster.ID'] <- 'DCLS Cluster ID'
colnames(crab_2020)[colnames(crab_2020) == 'WGS.Gene'] <- 'Resistance Gene'

#Creates Genes per cluster table (simplified version)
table_8 <- crab_2020 %>% 
  select('DCLS Cluster ID', 'Resistance Gene') %>%
  tbl_cross(row = 'DCLS Cluster ID',
            col = 'Resistance Gene',
            #percent = "row",
            statistic = NULL)%>%
  bold_labels()
table_8
```
\newpage
\noindent\rule{16cm}{0.4pt}
### 2021
\noindent\rule{16cm}{0.4pt}

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all', out.width="100%"}
#Divide and count CRAB sequencing total by year 2021
crab_2021 <- alldata_sep %>%
  filter(Year == 2021)

#Selecting only 2021 data and Graphing Clusters
crab_2021 %>% 
  count(Submitter, Cluster.ID) %>% 
  ggplot() +
  geom_col(mapping = aes(y = Submitter, fill =factor(Cluster.ID, levels=c("VA_AB_OXA23OXA24_231117","VA_AB_OXA23_230424", "VA_AB_OXA24_230526", "VA_AB_OXA24_231024", "VA_AB_OXA23_220720", "VA_AB_OXA23_230626", "VA_AB_OXA23_230424-1",  "VA_AB_OXA23NDM_240106", "VA_AB_OXA23_221103", "VA_AB_OXA23_230424-2", "VA_AB_OXA23_240205", "VA_AB_OXA23_230125", "VA_AB_OXA24_231119", "VA_AB_OXA24_220914 \n or VA_AB_OXA24_220120")), x = n), color = "black") +
  scale_fill_manual(values=Cluster_Palette) +
  scale_x_continuous(expand = c(0,0)) +
  xlab("Number of Isolates") +
  labs(title = "2021 Clusters per Submitter") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  legend.key.size = unit(.5, 'cm'),
    legend.text=element_text(size=8)) +
  guides(fill=guide_legend(title="DCLS Cluster ID")) #Rename legend
```

\newpage
Table 9: 2021 OXA Genes Identified Per Cluster
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "center", tab.cap = "2021 OXA Genes Identified Per Cluster", tab.id = "tab 9", label = "tab 9"}

#Genes per cluster table (old table too complicated)
#crab_2021 %>% 
#  select(Cluster.ID, WGS.Gene) %>% 
#  tbl_summary(by = WGS.Gene) %>%
#  modify_header(label = "**Cluster**") %>% 
#  add_overall() %>% 
#  as_flex_table() %>% 
#  flextable::fontsize(size = 7) %>% 
#  fontsize(size = 7, part = "header") %>% 
#  width(j = NULL, .85, unit = "in") %>%
#  autofit()

#Changes the Column names to be table ready
colnames(crab_2021)[colnames(crab_2021) == 'Cluster.ID'] <- 'DCLS Cluster ID'
colnames(crab_2021)[colnames(crab_2021) == 'WGS.Gene'] <- 'Resistance Gene'

#Creates Genes per cluster table (simplified version)
table_9 <- crab_2021 %>% 
  select('DCLS Cluster ID', 'Resistance Gene') %>%
  tbl_cross(row = 'DCLS Cluster ID',
            col = 'Resistance Gene',
            #percent = "row",
            statistic = NULL)%>%
  bold_labels()
table_9
```
\newpage
\noindent\rule{16cm}{0.4pt}
### 2022
\noindent\rule{16cm}{0.4pt}

```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Divide and count CRAB sequencing total by year 2022
crab_2022 <- alldata_sep %>%
  filter(grepl('2022EP', WGS.ID.))
crab_2022
#Selecting only 2022 data and Graphing Clusters
crab_2022 %>% 
  count(Submitter, Cluster.ID) %>% 
  ggplot() +
  geom_col(mapping = aes(y = Submitter, fill = factor(Cluster.ID, levels=c("VA_AB_OXA23OXA24_231117","VA_AB_OXA23_230424", "VA_AB_OXA24_230526", "VA_AB_OXA24_231024", "VA_AB_OXA23_220720", "VA_AB_OXA23_230626", "VA_AB_OXA23_230424-1",  "VA_AB_OXA23NDM_240106", "VA_AB_OXA23_221103", "VA_AB_OXA23_230424-2", "VA_AB_OXA23_240205", "VA_AB_OXA23_230125", "VA_AB_OXA24_231119", "VA_AB_OXA24_220914 \n or VA_AB_OXA24_220120")), x = n), 
           color = "black") +
  scale_fill_manual(values=Cluster_Palette) +
  xlab("Number of Isolates") +
  labs(title = "2022 Clusters per Submitter") +
  scale_x_continuous(expand = c(0,0)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  legend.key.size = unit(.5, 'cm'),
    legend.text=element_text(size=9),
     plot.title = element_text(hjust = 0)) +
  guides(fill=guide_legend(title="DCLS Cluster ID")) #Rename legend
```

\newpage
Table 10: 2022 OXA Genes Identified Per Cluster
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "left", tab.cap = "2022 OXA Genes Identified Per Cluster", tab.id = "tab 10", label = "tab 10"}
#Genes per Cluster table
#crab_2022 %>% 
#  select(Cluster.ID, WGS.Gene) %>% 
#  tbl_summary(by = WGS.Gene) %>%
#  modify_header(label = "**Cluster ID**") %>% 
#  add_overall() %>% 
#  as_flex_table() %>% 
#  flextable::fontsize(size = 7) %>% 
#  fontsize(size = 7, part = "header") %>% 
#  width(j = 1, 1.5, unit = "in") %>%
#  width(j = 2:6, .7, unit = "in")

#Changes the Column names to be table ready
colnames(crab_2022)[colnames(crab_2022) == 'Cluster.ID'] <- 'DCLS Cluster ID'
colnames(crab_2022)[colnames(crab_2022) == 'WGS.Gene'] <- 'Resistance Gene'

#Creates Genes per cluster table (simplified version)
table_10 <- crab_2022 %>% 
  select('DCLS Cluster ID', 'Resistance Gene') %>%
  tbl_cross(row = 'DCLS Cluster ID',
            col = 'Resistance Gene',
            #percent = "row",
            statistic = NULL)%>%
  bold_labels()
table_10
```
\newpage
\noindent\rule{16cm}{0.4pt}
### 2023
\noindent\rule{16cm}{0.4pt}
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide',fig.keep='all'}
#Divide and count CRAB sequencing total by year 2023
crab_2023 <- alldata_sep %>%
  filter(grepl('2023EP', WGS.ID.))
crab_2023
#Selecting only 2022 data and Graphing Clusters
crab_2023 %>% 
  count(Submitter, Cluster.ID) %>% 
  ggplot() +
  geom_col(mapping = aes(y = Submitter, fill = factor(Cluster.ID, levels=c("VA_AB_OXA23OXA24_231117","VA_AB_OXA23_230424", "VA_AB_OXA24_230526", "VA_AB_OXA24_231024", "VA_AB_OXA23_220720", "VA_AB_OXA23_230626", "VA_AB_OXA23_230424-1",  "VA_AB_OXA23NDM_240106", "VA_AB_OXA23_221103", "VA_AB_OXA23_230424-2", "VA_AB_OXA23_240205", "VA_AB_OXA23_230125", "VA_AB_OXA24_231119", "VA_AB_OXA24_220914 \n or VA_AB_OXA24_220120")), x = n), 
           color = "black") +
  scale_fill_manual(values=Cluster_Palette) +
  xlab("Number of Isolates") +
  labs(title = "2023 Clusters per Submitter") +
  scale_x_continuous(expand = c(0,0)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line = element_line(colour = "black"),
  legend.key.size = unit(.5, 'cm'),
    legend.text=element_text(size=9),
     plot.title = element_text(hjust = 0)) +
  guides(fill=guide_legend(title="DCLS Cluster ID")) #Rename legend
```

\newpage
Table 11: 2023 OXA Genes Identified Per Cluster
```{r, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, ft.align = "left", tab.cap = "2023 OXA Genes Identified Per Cluster", tab.id = "tab 11", label = "tab 11"}
#Genes per Cluster table (old table too complicated)
#crab_2023 %>% 
#  select(Cluster.ID, WGS.Gene)# %>% 
#  tbl_summary(by = WGS.Gene) %>%
#  modify_header(label = "**Cluster ID**") %>% 
#  add_overall() %>% 
#  as_flex_table() %>% 
#  flextable::fontsize(size = 7) %>% 
#  fontsize(size = 7, part = "header") %>% 
#  width(j = 1, 1.5, unit = "in") %>%
#  width(j = 2:8, .65, unit = "in") 

#Changes the Column names to be table ready
colnames(crab_2022)[colnames(crab_2022) == 'Cluster.ID'] <- 'DCLS Cluster ID'
colnames(crab_2022)[colnames(crab_2022) == 'WGS.Gene'] <- 'Resistance Gene'

#Creates Genes per cluster table (simplified version)
table_11 <- crab_2022 %>% 
  select('DCLS Cluster ID', 'Resistance Gene') %>%
  tbl_cross(row = 'DCLS Cluster ID',
            col = 'Resistance Gene',
            #percent = "row",
            statistic = NULL) %>%
  bold_labels()
table_11
```
\noindent\rule{16cm}{0.4pt}
##### Report Generated: `r Sys.Date()`
